integrations/stackit/src/haystack_integrations/components/embedders/stackit/document_embedder.py

```diff
@@ -78,4 +91,32 @@ def __init__(
             progress_bar=progress_bar,
             meta_fields_to_embed=meta_fields_to_embed,
             embedding_separator=embedding_separator,
+            timeout=timeout,
+            max_retries=max_retries,
+            http_client_kwargs=http_client_kwargs,
+        )
+        # We add these since they were only added in Haystack 2.14.0
+        self.timeout = timeout
+        self.max_retries = max_retries
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serializes the component to a dictionary.
+        :returns:
+            Dictionary with serialized data.
+        """
+        return default_to_dict(
+            self,
+            model=self.model,
+            api_key=self.api_key.to_dict(),
+            api_base_url=self.api_base_url,
+            prefix=self.prefix,
+            suffix=self.suffix,
+            batch_size=self.batch_size,
+            progress_bar=self.progress_bar,
+            meta_fields_to_embed=self.meta_fields_to_embed,
+            embedding_separator=self.embedding_separator,
+            timeout=self.timeout,
+            max_retries=self.max_retries,
+            http_client_kwargs=self.http_client_kwargs,
         )
```

Comment:
A question to better understand: can we remove this part after releasing 2.14.0?
Or do you plan to keep it?

Reply:
No we need it since otherwise we save variables like `dimensions` and `organization` which are not in the init signature of this component which then causes a deserialization error when running `from_dict` directly after a `to_dict`.